server/status_test.go

```diff
@@ -406,45 +405,60 @@ func TestStatusLocalLogs(t *testing.T) {
 
 	for i, testCase := range testCases {
 		var url bytes.Buffer
-		fmt.Fprintf(&url, "%s?level=%s&", "/_status/logs/local", testCase.Level.Name())
+		fmt.Fprintf(&url, "/_status/logs/local?level=%s", testCase.Level.Name())
 		if testCase.MaxEntities > 0 {
-			fmt.Fprintf(&url, "max=%d&", testCase.MaxEntities)
+			fmt.Fprintf(&url, "&max=%d", testCase.MaxEntities)
 		}
 		if testCase.StartTimestamp > 0 {
-			fmt.Fprintf(&url, "starttime=%d&", testCase.StartTimestamp)
+			fmt.Fprintf(&url, "&starttime=%d", testCase.StartTimestamp)
 		}
 		if testCase.StartTimestamp > 0 {
-			fmt.Fprintf(&url, "endtime=%d&", testCase.EndTimestamp)
+			fmt.Fprintf(&url, "&endtime=%d", testCase.EndTimestamp)
 		}
 		if len(testCase.Pattern) > 0 {
-			fmt.Fprintf(&url, "pattern=%s&", testCase.Pattern)
+			fmt.Fprintf(&url, "&pattern=%s", testCase.Pattern)
 		}
 
-		body = getRequest(t, ts, url.String())
-		entities := entryWrapper{}
-		if err := json.Unmarshal(body, &entities); err != nil {
+		var log logWrapper
+		path := url.String()
+		if err := json.Unmarshal(getRequest(t, ts, path), &log); err != nil {
 			t.Fatal(err)
 		}
-		var actualInfo, actualWarning, actualError bool
-		for _, entity := range entities.Data {
-			switch entity.Format {
-			case "TestStatusLocalLogFile test message-Error":
-				actualError = true
-			case "TestStatusLocalLogFile test message-Warning":
-				actualWarning = true
-			case "TestStatusLocalLogFile test message-Info":
-				actualInfo = true
+
+		if testCase.MaxEntities > 0 {
+			if a, e := len(log.Data), testCase.MaxEntities; a != e {
+				t.Errorf("%d expected %d entries, got %d: \n%+v", i, e, a, log.Data)
+			}
+		} else {
+			var actualInfo, actualWarning, actualError bool
+			var formats bytes.Buffer
+			for _, entry := range log.Data {
+				fmt.Fprintf(&formats, "%s\n", entry.Format)
+
+				switch entry.Format {
+				case "TestStatusLocalLogFile test message-Error":
+					actualError = true
+				case "TestStatusLocalLogFile test message-Warning":
+					actualWarning = true
+				case "TestStatusLocalLogFile test message-Info":
+					actualInfo = true
+				}
 			}
-		}
 
-		if testCase.ExpectedError != actualError {
-			t.Errorf("%d: expected error:%t, actual error:%t", i, testCase.ExpectedError, actualError)
-		}
-		if testCase.ExpectedWarning != actualWarning {
-			t.Errorf("%d: expected warning:%t, actual warning:%t", i, testCase.ExpectedWarning, actualWarning)
-		}
-		if testCase.ExpectedInfo != actualInfo {
-			t.Errorf("%d: expected info:%t, actual info:%t", i, testCase.ExpectedInfo, actualInfo)
+			if !(testCase.ExpectedInfo == actualInfo &&
```

Comment:
the three test cases don't provide a useful error message; they just tell you something failed but not why or what to do about it.